src/adapters/sumer/execution/src/sdk/lib.ts

```diff
@@ -0,0 +1,83 @@
+import { UserTVLData, Response } from "./types";
+import { JsonRpcProvider } from "ethers";
+import BigNumber from "bignumber.js";
+import path from "path";
+import { fetchGraphQLData } from "./fetch";
+require("dotenv").config({ path: path.join(__dirname, "../../.env") });
+
+const getAllUserPosition = async (blockNumber: number) => {
+  let result: UserTVLData[] = [];
+  let skip = 0;
+  const pageSize = 100;
+  let fetchNext = true;
+
+  while (fetchNext) {
+    const query = `
+      query MyQuery(
+        $skip: Int = ${skip},
+        $first: Int = ${pageSize},
+        $number: Int = ${blockNumber}
+      ) {
+        accountCTokens(first: $first, skip: $skip,  block: {number: $number}) {
+          cTokenBalance
+          market {
+            id
+            decimals
+            exchangeRate
+            underlyingAddress
+            underlyingSymbol
+          }
+          account {
+            id
+          }
+        }
+      }`;
+
+    const data = await fetchGraphQLData<Response>(query);
+
+    const { accountCTokens } = data;
+
+    const timestamp = await getTimestampAtBlock(blockNumber)
+    
+    const res = accountCTokens.map((data) => {
+      const userAddress = data.account.id;
+      const poolAddress = data.market.id;
+      const tokenAddress = data.market.underlyingAddress;
+      // const symbol = data.market.underlyingSymbol;
+      const balance = new BigNumber(data.cTokenBalance).times(`1e${data.market.decimals}`).times(data.market.exchangeRate).toFixed(0)
```

Comment:
Hi @yanfeizuo 
I understand that cTokenBalance and underlyingToken balance may not always be 1:1. This is because the underlyingToken may be affected by lending or other events, leading to a situation where cTokenBalance does not equal the underlyingToken balance. Our points are distributed based on the underlyingToken balance. According to my experience and understanding, the balance should be calculated using the following formula:

cTokenBalance / totalSupply * getCash()

Here, getCash() is the total amount of underlyingToken balance locked in the contract, totalSupply is the total number of cTokens minted, and cTokenBalance is the user's amount of cTokens. I hope my explanation is clear enough

Reply:
But, in fact, the result calculated by this formula is not the user's underlying balance.

Reply:
@yanfeizuo Isn't getCash the underlying balance locked in the contract?

Reply:
We should calculate the user's cToken percentage and multiply it by the underlying token amount held by the contract to determine the user's points balance.

Reply:
Will repull at once